src/Components/Components/src/ComponentFactory.cs

```diff
@@ -56,16 +56,22 @@ private void PerformPropertyInjection(IServiceProvider serviceProvider, ICompone
         private Action<IServiceProvider, IComponent> CreateInitializer([DynamicallyAccessedMembers(Component)] Type type)
         {
             // Do all the reflection up front
-            var injectableProperties =
-                MemberAssignment.GetPropertiesIncludingInherited(type, _injectablePropertyBindingFlags)
-                .Where(p => p.IsDefined(typeof(InjectAttribute)));
+            List<(string name, Type propertyType, PropertySetter setter)>? injectables = null;
+            foreach (var property in MemberAssignment.GetPropertiesIncludingInherited(type, _injectablePropertyBindingFlags))
```

Comment:
Might not matter, but as long as you're getting rid of LINQ usage, GetPropertiesIncludingInherited returns an IEnumerable that's the result of doing `dictionary.Values.SelectMany(p => p)`, and it looks like that method only has one other call site.  You might consider changing it to just return the dictionary, at which point this code can iterate through the dictionary (avoiding the Values collection creation and avoiding the enumerator allocation), and then have a nested loop for looping through each value's `List<Property>`, avoiding the `SelectMany`, its enumerable/enumerator, and a boxed enumerator for every list.

Reply:
The method is used in a few places which makes updating all of the call sites is a bit iffy. But I was able to update it to make it less-allocatey.